tests/unit/pages/Login.spec.js

```diff
@@ -0,0 +1,314 @@
+
+    await flushPromises();
+
+    expect(verifyOtp).toHaveBeenCalled();
+    expect(getUserByAccessToken).toHaveBeenCalled();
+    expect(mockRouter.replace).toHaveBeenCalledWith({
+      name: "Home",
+      params: {},
+    });
+  });
+
+  it("redirects to appropriate page after login", async () => {
+    const mockRouter = {
+      replace: jest.fn(),
+    };
+    const wrapper = mount(Login, {
+      global: {
+        mocks: {
+          $router: mockRouter,
+        },
+      },
+      props: {
+        redirectTo: "Editor",
+        params: '{"org":"","plioId":"abc"}',
+      },
+    });
+
+    // mock function to verify OTP
+    const verifyOtp = jest
+      .spyOn(UserAPIService, "verifyOtp")
+      .mockImplementation(() => {
+        return new Promise((resolve) => {
+          resolve({ data: dummyAccessToken });
+        });
+      });
+
+    // mock function to get user by access token
+    const getUserByAccessToken = jest
+      .spyOn(UserAPIService, "getUserByAccessToken")
+      .mockImplementation(() => {
+        return new Promise((resolve) => {
+          resolve({ data: dummyUser });
+        });
+      });
+
+    // enter valid number
+    await wrapper
+      .find('[data-test="phone"]')
+      .find('[data-test="input"]')
+      .setValue("9191919191");
+    // request OTP
+    await wrapper.find('[data-test="requestOTP"]').trigger("click");
+    // enter valid otp
+    await wrapper
+      .find('[data-test="otp"]')
+      .find('[data-test="input"]')
+      .setValue("123456");
+    // submit otp
+    await wrapper.find('[data-test="submitOTP"]').trigger("click");
+
+    await flushPromises();
+
+    expect(verifyOtp).toHaveBeenCalled();
+    expect(getUserByAccessToken).toHaveBeenCalled();
+    expect(mockRouter.replace).toHaveBeenCalledWith({
+      name: "Editor",
+      params: {
+        org: "",
+        plioId: "abc",
+      },
+    });
+  });
+
+  it("resends OTP", async () => {
+    const wrapper = mount(Login);
+
+    // mock function to verify OTP
+    const requestOtp = jest
+      .spyOn(UserAPIService, "requestOtp")
+      .mockImplementation(() => jest.fn());
+
+    // enter valid number
+    await wrapper
+      .find('[data-test="phone"]')
+      .find('[data-test="input"]')
+      .setValue("9191919191");
+    // request OTP
+    await wrapper.find('[data-test="requestOTP"]').trigger("click");
+    // resend OTP
+    await wrapper.find('[data-test="resendOTP"]').trigger("click");
+
+    await flushPromises();
+
+    expect(requestOtp).toHaveBeenCalled();
+    expect(wrapper.vm.resentOtp).toBe(true);
+    expect(wrapper.vm.invalidOtp).toBe(false);
+  });
+
+  it("does not log in with Google when null user", async () => {
+    const wrapper = mount(Login);
+
+    // google button should become enabled in a little time
+    await new Promise((r) => setTimeout(r, 200));
+
+    await wrapper.find('[data-test="googleLogin"]').trigger("click");
+    expect(wrapper.vm.isGoogleAuthDisabled).toBe(false);
```

Comment:
it should stay disabled right? here the check is written for the google auth to be enabled